packages/palette/src/elements/Carousel/Carousel.tsx

```diff
@@ -180,6 +182,10 @@ export const Carousel: React.FC<CarouselProps> = ({
     onChange && onChange(index)
   }, [onChange, index])
 
+  useEffect(() => {
+    onPageCountChange && onPageCountChange(pages.length)
+  }, [onPageCountChange, pages.length])
```

Comment:
Should this be running on initial mount as well? (See the above `useUpdateEffect` which skips that.

Reply:
Yea, make sense. missed the hook.